modules/sdp/SDP.js

```diff
@@ -24,7 +24,30 @@ export default class SDP {
      * @param {boolean} isP2P - Whether this SDP belongs to a p2p peerconnection.
      */
     constructor(sdp, isP2P = false) {
-        const media = sdp.split('\r\nm=');
+        this._updateSessionAndMediaSections(sdp);
+        this.isP2P = isP2P;
+        this.raw = this.session + this.media.join('');
```

Comment:
Question: would it make sense to make `raw ` a getter which creates its value when accessed? Looks like we need to keep it in sync with `session` and `media`, but a getter would help simplify that. The downside would be if we access it many many times, since its value would be computed every time.

Reply:
I started off by introducing a getter for raw but it is indeed being accessed many times in JIngleSessionPC resuting in unnecessary recomputations. I did run into another issue as well where the SDP was getting messed up.

Reply:
Isn't it mostly being accessed _after_ being computed? In that case we'd have the same amount of computations, but without the risk of missing updating raw when we updated something else.

Perhaps we can look into it after this PR.

Reply:
No, there are instances where it is simply passed to sdp-transofrm for parsing and also for renegotiations. I agree that optimizing this could be a separate PR.